sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -705,7 +705,8 @@ void CurlSession::ReadStatusLineAndHeadersFromRawResponse(
   // For NoContent status code, also need to set contentLength to 0.
   // https://github.com/Azure/azure-sdk-for-cpp/issues/406
   if (this->m_request.GetMethod() == HttpMethod::Head
-      || this->m_lastStatusCode == HttpStatusCode::NoContent)
+      || this->m_lastStatusCode == HttpStatusCode::NoContent
+      || this->m_lastStatusCode == HttpStatusCode::NotModified)
```

Comment:
This bug fix needs a unit test and a changelog entry.

@Jinming-Hu What other status codes do we need to check for? Why is this one special, and is that storage specific or general for all services?

Reply:
> What other status codes do we need to check for?

I don't know, we need to check HTTP specification to answer.

> Why is this one special

204 and 304 are status codes storage service will return.

>  is that storage specific or general for all services?

all services.